File: 2470-removing-stars-from-a-string/2470-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
     Stack<Character> result=new Stack<>();
     StringBuilder sb=new StringBuilder();
     for(int i=0;i<s.length();i++){
        if(s.charAt(i) != '*'){
            result.push(s.charAt(i));
        }
        else{
            result.pop();
        }
     }   
     for(char c : result){
        sb.append(c);
     }
     return sb.toString();
    }
}

File: 2470-removing-stars-from-a-string/2470-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        int n = s.size();
        stack<char> st;
        for(int i = 0; i < n; i++)
        {
            st.push(s[i]);
            if(s[i] == '*')
            {
                st.pop();
                st.pop();
            }
        }
        string ans = "";
        while(! st.empty())
        {
            ans += st.top();
            st.pop();
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};